robyn/__init__.py

```diff
@@ -191,12 +191,19 @@ def startup_handler(self, handler: Callable) -> None:
     def shutdown_handler(self, handler: Callable) -> None:
         self._add_event_handler(Events.SHUTDOWN, handler)
 
-    def is_port_in_use(self, port: int) -> bool:
+    def get_socket(self, url: str, port: int) -> (bool, SocketHeld):
+        """
+        @param url: the host URL
+        @param port: the port number
+        @return: a tuple containing a bool that denotes whether
+        the socket was acquired or not followed by the actual socket, if successful
+        """
         try:
             with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
-                return s.connect_ex(("localhost", port)) == 0
+                return s.connect_ex(("localhost", port)) != 0, SocketHeld(url, port)
```

Comment:
@VishnuSanal , one final suggestion. Split this into two lines

i.e.

```
acquired_socket = SocketHeld(url, port)
is_valid = s.connect_ex()
```

Also, now that I think of it. we are only checking "localhost" here. But we also deploy to "0.0.0.0" . could you use a variable `host` here instead?

Reply:
Hey @VishnuSanal , just some final suggestions 👋